yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -231,6 +249,17 @@ private[yarn] class YarnAllocator(
   }
 
   /**
+   * Creates a container request, handling the reflection required to use YARN features that were
+   * added in recent versions.
+   */
+  private def createContainerRequest(resource: Resource): ContainerRequest = {
+    nodeLabelConstructor.map { constructor =>
```

Comment:
Just reviving my previous question: should we blow up if the user requests a node label but the library doesn't support the feature?